byoqm/source_repository/source_repository.py

```diff
@@ -82,17 +84,29 @@ def _parse_ast(self, of_file: FileInfo) -> tree_sitter.Tree:
             return ast
 
     def _discover_files(self) -> Dict[Path, FileInfo]:
+        ignored_files = self._get_ignored_files()
+
         if self.src_root.is_file():
+            if self.src_root in ignored_files:
+                logging.warning(
+                    "Source directory is a file and is included in the .ignore file"
+                )
```

Comment:
I don't think this should be a warning because there is a lot of intentionality on the user's side for this to be ignored - maybe an info level log is sufficient?